[2024-06-14T16:35:28.966+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2024-06-14T16:35:29.516+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: Air_pollution_data_pipeline.avg_emission_by_year scheduled__2024-05-30T00:00:00+00:00 [queued]>
[2024-06-14T16:35:29.614+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: Air_pollution_data_pipeline.avg_emission_by_year scheduled__2024-05-30T00:00:00+00:00 [queued]>
[2024-06-14T16:35:29.616+0000] {taskinstance.py:2306} INFO - Starting attempt 6 of 6
[2024-06-14T16:35:31.177+0000] {taskinstance.py:2330} INFO - Executing <Task(_PythonDecoratedOperator): avg_emission_by_year> on 2024-05-30 00:00:00+00:00
[2024-06-14T16:35:31.351+0000] {standard_task_runner.py:63} INFO - Started process 1831 to run task
[2024-06-14T16:35:31.414+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'Air_pollution_data_pipeline', 'avg_emission_by_year', 'scheduled__2024-05-30T00:00:00+00:00', '--job-id', '361', '--raw', '--subdir', 'DAGS_FOLDER/main.py', '--cfg-path', '/tmp/tmpnr0j0fxn']
[2024-06-14T16:35:31.445+0000] {standard_task_runner.py:91} INFO - Job 361: Subtask avg_emission_by_year
[2024-06-14T16:35:35.246+0000] {task_command.py:426} INFO - Running <TaskInstance: Air_pollution_data_pipeline.avg_emission_by_year scheduled__2024-05-30T00:00:00+00:00 [running]> on host 6078ceb33835
[2024-06-14T16:35:38.022+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='khalid' AIRFLOW_CTX_DAG_ID='Air_pollution_data_pipeline' AIRFLOW_CTX_TASK_ID='avg_emission_by_year' AIRFLOW_CTX_EXECUTION_DATE='2024-05-30T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='6' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2024-05-30T00:00:00+00:00'
[2024-06-14T16:35:38.048+0000] {taskinstance.py:430} INFO - ::endgroup::
[2024-06-14T16:35:38.060+0000] {summary_analysis.py:40} INFO - starting average pollutants emission calculation
[2024-06-14T16:35:38.133+0000] {logging_mixin.py:188} WARNING - /opt/***/dags/statistical_analysis/summary_analysis.py:43 FutureWarning: Passing literal json to 'read_json' is deprecated and will be removed in a future version. To read from a literal string, wrap it in a 'StringIO' object.
[2024-06-14T16:35:38.198+0000] {logging_mixin.py:188} WARNING - /opt/***/dags/statistical_analysis/summary_analysis.py:45 FutureWarning: The behavior of obj[i:j] with a float-dtype index is deprecated. In a future version, this will be treated as positional instead of label-based. For label-based slicing, use obj.loc[i:j] instead
[2024-06-14T16:35:38.208+0000] {summary_analysis.py:53} ERROR - error calculating average emission
Traceback (most recent call last):
  File "/opt/airflow/dags/statistical_analysis/summary_analysis.py", line 46, in avg_emission_by_year
    avg_emision = sub_data.groupby('Year').mean().reset_index()
  File "/home/airflow/.local/lib/python3.10/site-packages/pandas/core/frame.py", line 9183, in groupby
    return DataFrameGroupBy(
  File "/home/airflow/.local/lib/python3.10/site-packages/pandas/core/groupby/groupby.py", line 1329, in __init__
    grouper, exclusions, obj = get_grouper(
  File "/home/airflow/.local/lib/python3.10/site-packages/pandas/core/groupby/grouper.py", line 1043, in get_grouper
    raise KeyError(gpr)
KeyError: 'Year'
[2024-06-14T16:35:38.274+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2024-06-14T16:35:38.275+0000] {taskinstance.py:2905} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/baseoperator.py", line 401, in wrapper
    return func(self, *args, **kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/decorators/base.py", line 265, in execute
    return_value = super().execute(context)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/baseoperator.py", line 401, in wrapper
    return func(self, *args, **kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/operators/python.py", line 235, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/operators/python.py", line 252, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/opt/airflow/dags/statistical_analysis/summary_analysis.py", line 46, in avg_emission_by_year
    avg_emision = sub_data.groupby('Year').mean().reset_index()
  File "/home/airflow/.local/lib/python3.10/site-packages/pandas/core/frame.py", line 9183, in groupby
    return DataFrameGroupBy(
  File "/home/airflow/.local/lib/python3.10/site-packages/pandas/core/groupby/groupby.py", line 1329, in __init__
    grouper, exclusions, obj = get_grouper(
  File "/home/airflow/.local/lib/python3.10/site-packages/pandas/core/groupby/grouper.py", line 1043, in get_grouper
    raise KeyError(gpr)
KeyError: 'Year'
[2024-06-14T16:35:38.356+0000] {taskinstance.py:1206} INFO - Marking task as FAILED. dag_id=Air_pollution_data_pipeline, task_id=avg_emission_by_year, run_id=scheduled__2024-05-30T00:00:00+00:00, execution_date=20240530T000000, start_date=20240614T163529, end_date=20240614T163538
[2024-06-14T16:35:38.428+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 361 for task avg_emission_by_year ('Year'; 1831)
[2024-06-14T16:35:38.506+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 1
[2024-06-14T16:35:38.775+0000] {taskinstance.py:3503} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-06-14T16:35:38.866+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
